src/azul/chalice.py

```diff
@@ -467,6 +481,36 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
+    def threshold(self, *, errors: int, throttles: int):
+        def wrapper(f):
+            f.errors_threshold = errors
+            f.throttles_threshold = throttles
+            return f
+
+        return wrapper
+
+    def metric_thresholds(self) -> list[MetricThreshold]:
+        default_threshold = 0
+        thresholds = []
+        lambda_name, _ = config.unqualified_resource_name(self.app_name)
+        for metric in MetricThreshold.metrics:
+            # The api_handler lambda functions (indexer & service) aren't included
+            # in the app_module's handler_map, so we account for those first.
+            thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                              metric=metric,
+                                              threshold=default_threshold))
+            for handler in self.handler_map.values():
+                if isinstance(handler, chalice.app.EventSourceHandler):
+                    func = handler.func
+                    if isinstance(handler.func, chalice.app.PureLambdaWrapper):
+                        func = handler.func._original_func
```

Comment:
I think you can eliminate this dependency on Chalice-internals by reordering the decorator invocations. The decorating function would then receive a handler instance for you to stash the thresholds in, rather than the actual handler function, and you wouldn't need to dig so deep here to get at the handler function.